src/states/node.rs

```diff
@@ -1366,7 +1393,10 @@ impl Node {
     }
 
     /// Returns `Ok` if a client is allowed to send the given message.
-    fn check_valid_client_message(&self, msg: &RoutingMessage) -> Result<(), RoutingError> {
+    fn check_valid_client_message(&mut self,
```

Comment:
shall we use part of the work in https://github.com/Fraser999/routing/pull/4/commits/6d4781012265f57d1685da37986a232777da8eb2
to move this `check_valid_client_message` function into RateLimit
so node.rs can be cleaned_up little bit?